packages/specs/schema/src/utils/mergeSchema.ts

```
@@ -0,0 +1,126 @@
+    }
+
+    return acc;
+  }, {} as Properties);
+}
+
+function mergeManyOf(kind: "allOf" | "anyOf" | "oneOf", schema1: JSONSchema7, schema2: JSONSchema7) {
+  const refs = new Set();
+
+  if (schema1.$ref && schema2.type && (schema2.readOnly || schema2.writeOnly) && kind === "allOf") {
+    const {type, ...rest} = schema2;
+    schema2 = rest;
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)